sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -546,43 +546,6 @@ joint_matrix<sub_group, int, use::accumulator, msize, nsize> sub_c;
 //Remainder handling
 ```
 
-## Future-looking API
-
-### Memory scope
-The current experimental API uses `joint_` semantics to define the memory scope of the matrix. The long term solution is to use the proposed link:../supported/sycl_ext_oneapi_local_memory.asciidoc[`group_local_memory` extension] to allocate the matrix in local memory associated with a SYCL group as shown in the example below.
-
-
-```c++
-multi_ptr<matrix<T>, address_space::local_space> tA_ptr = group_local_memory<matrix<sub_group, int8_t, tM, tN, use::a>>(sg);
-```
-We did not utilize this extension for this matrix API version because sub-group local memory is not yet well defined in {dpcpp}. Moreover, the representation of this notion in LLVM IR and SPIR-V is not clear yet. 
-
-### WI data to joint matrix mapping coordinates information for piece-wise operations
-The indexing provided inside the `wi_data` class accesses only the portion of the matrix held by the current WI. It is not possible to know the location of this portion in the original matrix.  This coordinates mapping  is implementation defined and changes from one backend to the other. For general piece-wise operations like sum of rows of a matrix, the WI data to joint matrix mapping information is needed to reason about the matrix view.
-Within the joint matrix extension, we want to write, as much as possible, one code to run on different backends. If backend X states that a WI owns one exact row of the matrix for instance, writing the following code will work only on that backend for that version of hardware. If a different hardware and implementation is used, the same WI may own only half of the row if, for example, the SG size increased. 
-
-```c++
-auto data = get_wi_data(sg, C);
-for (int i = 0; i < data.length(); ++i) {
-  sum_of_local_rows[row] += data[i];
-}
-```
-
-We want to keep backward compatibility in the joint matrix code when implementations or hardware change. To that end, instead of hard-coding this mapping, we use general backend and target-agnostic functionality, especially in the JIT compilation mode of SYCL. For this reason we would like to be able to query this mapping so that code does not have to change from one version to the other.
-
-So for the mapping problem, since this mapping is implementation-defined, one of the proposals is to add runtime functions like:
-```c++
-auto data = get_wi_data(sg, C);
-for (int i = 0; i < data.length; ++i) {
-  auto row, col = data[i].get_coord();
-  sum_of_local_rows[row] += data[i];
-}
-```
-
-## TODO List
-- Add WI data to joint matrix mapping coordinates information for piece-wise operations. This will be added as part of the query or new methods to the 'get_wi_data' class. 
-- Add a more realistic and complete example that shows the value of the general query. 
```

Comment:
@gmlueck, @JackAKirk,  I added an example that shows the general query API in https://github.com/intel/llvm-test-suite/pull/1492, In SYCL/Matrix/joint_matrix_query_general_tpu_sel.cpp.
This example employs: combinations, `max_msize`, `msize`, `atype`, `btype`, `accumulatortype`, and `scopes`.
 All these are needed to query a specific combination on both AMX and XMX.
I don't see a field that can be left out for both functionality and performance.
The only field I don't show in the other PR example is `num_tiles` for AMX. This is also important and was used in the AMX performance kernels. The idea is illustrated here. In order to decide how  many joint matrices (AMX tiles) to use in the innermost loop body, the user needs to know that the maximum is 8 on AMX. Otherwise you end up with spills and bad performance.  In these performance kernels, one option is to use the following code that uses total of 8 tiles but with tile reuse.  In pseudo code:
// 4 C tiles
joint_matrix<> C00;
joint_matrix<> C01;
joint_matrix<> C10;
joint_matrix<> C11;
for k loop
// 2 A tiles
  joint_matrix<> A0;
  joint_matrix<> A1;
// 2 B tiles
  joint_matrix<> B0;
  joint_matrix<> B1;
//loads of two A and two B
  C00 = joint_matrix_mad(C00, A0, B0);
  C01 = joint_matrix_mad(C01, A0, B1);
  C10 = joint_matrix_mad(C10, A1, B0);
  C11 = joint_matrix_mad(C11, A1, B1);
} //end k loop
//stores of the 4 C matrices